src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java

```diff
@@ -42,4 +57,68 @@ public void getAuthorization(String batchId, String clientCorrelationId, Authori
             logger.error(e.toString());
         }
     }
+
+    public BatchDTO getBatchSummary(String batchId) {
+        return successfulBatchSummaryResponse(batchId);
+    }
+
+    public BatchDetailResponse getBatchDetails(String batchId, int pageNo, int pageSize) {
+        List<Transfer> transactions = getTransactions(batchId);
+        int toIndex = pageNo * pageSize;
+        int fromIndex = toIndex - pageSize;
+        toIndex = Math.min(toIndex, transactions.size());
+        BatchDetailResponse batchDetailResponse = new BatchDetailResponse();
+        batchDetailResponse.setContent(transactions.subList(fromIndex, toIndex));
+        return batchDetailResponse;
+    }
+
+    private BatchDTO successfulBatchSummaryResponse(String batchId) {
+        Long total = 10L;
+        Long ongoing = 1L;
+        Long failed = 1L;
+        Long successful = 8L;
+        BigDecimal totalAmount = BigDecimal.valueOf(100);
+        BigDecimal ongoingAmount = BigDecimal.valueOf(0);
+        BigDecimal failedAmount = BigDecimal.valueOf(10);
+        BigDecimal successfulAmount = BigDecimal.valueOf(90);
+        String status = "Pending";
+        String successPercentage = "90";
+        String failedPercentage = "10";
+
+        return new BatchDTO(batchId, null, total, ongoing, failed, successful, totalAmount, successfulAmount, ongoingAmount, failedAmount,
+                null, null, null, status, null, null, failedPercentage, successPercentage);
+    }
+
+    private List<Transfer> getTransactions(String batchId) {
+        List<Transfer> transactionList = new ArrayList<>();
+
+        for (int index = 0; index < 10; index++) {
+            Transfer transfer;
+
+            if (successTxnCount > 0) {
+                transfer = getSingleTransaction(index, ThreadLocalRandom.current().nextLong(), UUID.randomUUID().toString(),
+                        TransferStatus.COMPLETED, batchId);
+            } else {
+                transfer = getSingleTransaction(index, ThreadLocalRandom.current().nextLong(), UUID.randomUUID().toString(),
+                        TransferStatus.IN_PROGRESS, batchId);
+            }
+            transactionList.add(transfer);
+        }
+        return transactionList;
+    }
+
+    private Transfer getSingleTransaction(int index, Long workflowInstanceKey, String requestId, TransferStatus status, String batchId) {
+        String id = String.valueOf(index);
+        Date startedAt = new Date(1685536200000L);
```

Comment:
is there a test case available for this?

Reply:
No, manoj. All data is mocked.

Reply:
No, it will always send same data. No need for test case here